hopp/offshore/floating_platform.py

```diff
@@ -0,0 +1,345 @@
+        self.config = self.validate_config(config)
+
+        self.initialize_port()
+        self.setup_simulation(**kwargs)
+
+    # Setup simulation seems to be the install phase's equivalent run() module
+    def setup_simulation(self, **kwargs):
+
+        #print("Floating Platform Install setup_sim() is working!!!")
+
+        self.distance = self.config['site']['distance']
+        self.depth = self.config['site']['depth']
+        self.mass = self.config['equipment']['tech_combined_mass']
+        self.area = self.config['equipment']['tech_required_area']
+
+        _platform = self.config.get('equipment', {})
+        design_cost = _platform.get('topside_design_cost', 4.5e6)   # USD
+        fab_cost = _platform.get('fabrication_cost_rate', 14500.)   # USD/t
+        steel_cost = _platform.get('substructure_steel_cost', 3000) # USD/t
+        #NEED Updated version for floating
+        install_duration = _platform.get("install_duration", 14)    # days
+        
+        # Initialize vessel 
+        vessel_specs = self.config.get("oss_install_vessel", None)
+        name = vessel_specs.get("name","Offshore Substation Install Vessel")
+
+        vessel = Vessel(name, vessel_specs)
+        self.env.register(vessel)
+
+        vessel.initialize()
+        self.install_vessel = vessel
+        
+        # Add in the mass of the substructure to total mass (may or may not impact the final install cost)
+        _, substructure_mass = calc_substructure_mass_and_cost(self.mass, self.area, 
+                        self.depth, fab_cost, design_cost, steel_cost
+                        )
+
+        total_mass = self.mass + substructure_mass  # t
+
+         # Call the install_platform function
+        self.install_capex = install_platform(total_mass, self.area, self.distance, \
+                                                   install_duration, self.install_vessel)
+
+    # An install object needs to have attribute system_capex, installation_capex, and detailed output
+    @property
+    def system_capex(self):
+
+        return {}
+
+    @property 
+    def installation_capex(self):
+        
+        return self.install_capex
+
+    @property
+    def detailed_output(self):
+
+        return {}
+
+# Define individual calculations and functions to use outside or with ORBIT
+def calc_substructure_mass_and_cost(mass, area, depth, fab_cost=14500., design_cost=4.5e6, sub_cost=3000, pile_cost=0):
+    '''
+    Platform is substructure and topside combined
+    All funstions are based off NREL's ORBIT (oss_design)
+    default values are specified in ORBIT
+    '''
+    #Inputs needed
+    topside_mass = mass
+    topside_fab_cost_rate   =   fab_cost   
+    topside_design_cost     =   design_cost
+
+    '''Topside Cost & Mass
+    Topside Mass is the required Mass the platform will hold
+    Topside Cost is a function of topside mass, fab cost and design cost'''
+    topside_cost   =   topside_mass   *topside_fab_cost_rate  +topside_design_cost
```

Comment:
This is a bit nit-picky, but it is helpful to have no spaces around multiplication and have spaces on both sides of addition and subtraction. This approach helps to group things visually the way they will be operated on. See https://peps.python.org/pep-0008/#whitespace-in-expressions-and-statements for more information.

Reply:
If you want to do this automatically, there is a tool called "black" that auto formats code according to PEP 8 standards. Probably does not make sense to include it in HOPP yet, but I'd like to eventually.